internal/api/server.go

```diff
@@ -0,0 +1,43 @@
+package api
+
+import (
+	"log"
+	"net/http"
+	"net/http/httptest"
+)
+
+const testAPI = "this-is-my-secure-token-do-not-steal!!"
+
+func GetTestToken() string {
+	return testAPI
+}
+
+// OpenAITestServer Creates a mocked OpenAI server which can pretend to handle requests during testing.
+func OpenAITestServer() *httptest.Server {
+	return httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		log.Printf("received request at path %q\n", r.URL.Path)
+
+		// check auth
+		if r.Header.Get("Authorization") != "Bearer "+GetTestToken() {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		handler, ok := serverMap[r.URL.Path]
+		if !ok {
+			http.Error(w, "the resource path doesn't exist", http.StatusNotFound)
+			return
+		}
+		handler(w, r)
+	}))
+}
+
+type Handler func(w http.ResponseWriter, r *http.Request)
+
+// serverMap.
+var serverMap = make(map[string]Handler)
```

Comment:
Not sure that global map is a good idea here, maybe we can do something along the lines of:

```go
type TestServer struct {
	handlers map[string]Handler
}

func NewTestServer() *TestServer { ... }
func (ts *TestServer) RegisterHandler(path string, handler Handler) { ... }
```

Thanks!

Reply:
Should the map in RegisterHandler be determined whether it exists or directly overwritten?

Reply:
I think global map is feasible here, and there is no need to consider concurrency. Maybe I think it is simple, don't you think?

Reply:
We also have `GetTestToken` here, I think it makes sense to keep test-server-related logic encapsulated in a single tidy struct, in that way, it would be obvious how to extend the test server in the future.

Reply:
Well, I'll make it simple